pennylane/qchem/hartree_fock.py

```diff
@@ -125,17 +125,18 @@ def _scf(*args):
         charges = mol.nuclear_charges
         r = mol.coordinates
         n_electron = mol.n_electrons
+        argnums = [mol.coord_opt, mol.coeff_opt, mol.alpha_opt]
 
-        if r.requires_grad:
+        if getattr(r, "requires_grad", False) or argnums[0]:
```

Comment:
Do we still keep the `requires_grad` in the codebase or handle it via the `_opt` arguments? 
cc @soranjh

Reply:
I think we want to keep it backwards compatible right? if we remove requires_grad now, people who use pnp are going to be in for a surprise 😅

Reply:
We should keep the support for `requires_grad`.